libbeat/processors/actions/decode_json_fields.go

```diff
@@ -0,0 +1,93 @@
+package actions
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+	"github.com/pkg/errors"
+)
+
+type decodeJSONFields struct {
+	Fields []string
+}
+
+func init() {
+	processors.RegisterPlugin("decode_json_fields", configChecked(newDecodeJSONFields,
+		requireFields("fields"), allowedFields("fields", "when")))
+}
+
+func newDecodeJSONFields(c common.Config) (processors.Processor, error) {
+	config := struct {
+		Fields []string `config:"fields"`
+	}{}
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the decode_json_fields configuration: %s", err)
+	}
+
+	f := decodeJSONFields{Fields: config.Fields}
+	return f, nil
+}
+
+func (f decodeJSONFields) Run(event common.MapStr) (common.MapStr, error) {
+	errs := []string{}
+
+	for _, field := range f.Fields {
+		var output map[string]interface{}
```

Comment:
This doesn't need to be declared until your are actually going to use it. So move it inside of the `if ok { unmarshal ...` block.